src/cheri/riscv-unpriv-integration.adoc

```diff
@@ -375,6 +389,63 @@ The zero register is extended with zero metadata and a cleared valid tag: this i
 
 NOTE: Instruction encodings refer to *c* registers when they use `cs1`, `cd`, etc. instead of `rs1`, `rd` (which reads/writes the XLEN-bit subset of the register).
 
+==== Added CSRs
+
+{cheri_base32_ext_name} and {cheri_base64_ext_name} add the capability CSR shown in
+xref:utidc-csrnames-added[xrefstyle=short].
+
+[[utidc-csrnames-added]]
+.Unprivileged capability CSRs added in {cheri_base32_ext_name} and {cheri_base64_ext_name}
+[%autowidth,float="center",align="center",cols="<,<,<,<",options="header"]
+|===
+|CLEN CSR|Address|Permissions|Description
+|<<utidc>>|0x480|RW, <<asr_perm>> required for writes, not reads|User Default Data Capability
+|===
+
+[#utidc,reftext="utidc"]
+===== User Thread Identifier Capability (utidc)
+
+The <<utidc>> register is used to identify the current software thread in user mode.
+On reset the valid tag of <<utidc>> will be set to 0 and the remainder of the data is UNSPECIFIED.
+
+
+NOTE: Compartmentalization seeks to separate the privileges between different
+protection units, e.g., two or more libraries. Code can be separated by sentries,
+which allow for giving out code capabilities to untrusted code where the untrusted
+code can only call the code capability, but not modify it. The <<utidc>> register supports
+a model where untrusted code is separated by trusted code and each call from one piece of untrusted
+code to another piece of untrusted piece of code goes through trusted code. Often, the trusted
+code is referred to as a *trampoline*. Sentries can
+be called from different software threads and thus there needs to be a way of
+identifying the current software thread. While identifying the current software thread
+can be done by privileged code, e.g., the kernel, the implied performance
+overhead of this is not bearable for CHERI systems with many compartments.
+
+NOTE: The <<utidc>> register is designed to hold a capability, which can only be accessed by
+trusted code. In a commonly used model on CHERI systems, the trusted code's responsibility
+is only to switch between compartments, but not to switch threads. This responsibility is usually
+taken over by more privileged code, e.g., an operating system kernel running on a different privilege level.
+The privileged code switches software threads and writes the <<utidc>> register.
+
+NOTE: Every piece of code in the user space (and more privileged levels) can read the content of the <<utidc>>
```

Comment:
This is a lot of software guidance that is useful but probably shouldn't be part of this chapter.